core/sysparam.c

```diff
@@ -719,97 +761,65 @@ sysparam_status_t sysparam_get_string(const char *key, char **destptr) {
     return SYSPARAM_OK;
 }
 
-sysparam_status_t sysparam_get_int(const char *key, int32_t *result) {
-    char *buffer;
-    char *endptr;
+sysparam_status_t sysparam_get_int32(const char *key, int32_t *result) {
     int32_t value;
+    size_t actual_length;
+    bool is_binary;
     sysparam_status_t status;
 
-    status = sysparam_get_string(key, &buffer);
+    status = sysparam_get_data_static(key, (uint8_t *)&value, sizeof(int32_t),
+                                      &actual_length, &is_binary);
     if (status != SYSPARAM_OK) return status;
-    value = strtol(buffer, &endptr, 0);
-    if (*endptr) {
-        // There was extra crap at the end of the string.
-        free(buffer);
+    if (!is_binary || actual_length != sizeof(int32_t))
         return SYSPARAM_PARSEFAILED;
-    }
-
     *result = value;
-    free(buffer);
-    return SYSPARAM_OK;
+    return status;
 }
 
-sysparam_status_t sysparam_get_bool(const char *key, bool *result) {
-    char *buffer;
+sysparam_status_t sysparam_get_int8(const char *key, int8_t *result) {
```

Comment:
Part of the purpose of sysparam was also to create an easy way for the core and applications to read and write human-readable configuration settings for a lot of parameters that humans often want to change.  That is why both the bool routines and the int stuff was designed to work with human-readable strings.  (If you're concerned about size, note that bool values can be stored as single-byte"y"/"n"/etc, and in the case of ints, the majority of int values people are likely to be working with are actually smaller when stored as text than as a 32-bit binary value)

I can see some value in reducing the need for mallocs, but I think this is an unnecessary removal of useful behavior.  If you really want to do this I would recommend perhaps adding a way to read/write small strings internally using the bounce buffer so that bool and int text values could be read/written entirely using that buffer..